Source/JavaScriptCore/assembler/MacroAssemblerARM64.cpp

```diff
@@ -875,12 +875,21 @@ void MacroAssemblerARM64::collectCPUFeatures()
 #define HWCAP_ATOMICS (1 << 8)
 #endif
 
+#if !defined(HWCAP_FPHP)
+#define HWCAP_FPHP (1 << 9)
+#endif
+
+#if !defined(HWCAP_ASIMDHP)
+#define HWCAP_ASIMDHP (1 << 10)
+#endif
+
 #if !defined(HWCAP_JSCVT)
 #define HWCAP_JSCVT (1 << 13)
 #endif
 
         s_lseCheckState = (hwcaps & HWCAP_ATOMICS) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
         s_jscvtCheckState = (hwcaps & HWCAP_JSCVT) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+        s_float16CheckState = ((hwcaps & HWCAP_FPHP) && (hwcaps & HWCAP_ASIMDHP)) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
```

Comment:
nice, i know that people use Bun on raspberry pi's on linux arm64 that probably do not have this instruction

Reply:
Yeah, this is ARMv8.3 or later.